src/Core/Web/XMLHttpRequestHelper.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Generated On: 2015-10-5
+ * Class: XMLHttpRequestHelper. Simple promise-based wrapper around XMLHttpRequest
+ */
+
+
+define('Core/Web/XMLHttpRequestHelper', ['when'], function(when) {
+    return {
+        fetch: function(url, responseType) {
```

Comment:
I have no strong feeling regarding `fetch` vs `XmlHttpRequest` and `promise` vs `when.`

But:
  * for fetch: if we want to switch, this PR is a good opportunity to do it because it's focused primarly on XmlHttpRequest
  * for Promise: I'd prefer to wait do it in another PR where we would switch *all* `when` usage to Promise